core/server/policy_violations.go

```diff
@@ -0,0 +1,267 @@
+			}
+			for i := range list.Items {
+				validation, err := toPolicyValidation(list.Items[i], listClusterName, extraDetails)
+				if err != nil {
+					return nil, fmt.Errorf("error while getting policy violation event details: %w", err)
+				}
+				validations = append(validations, validation)
+			}
+		}
+	}
+
+	return &validationList{
+		Validations: validations,
+		Token:       clist.GetContinue(),
+		Errors:      respErrors,
+	}, nil
+}
+
+func toPolicyValidation(item v1.Event, clusterName string, extraDetails bool) (*pb.PolicyValidation, error) {
+	annotations := item.GetAnnotations()
+	policyValidation := &pb.PolicyValidation{
+		Id:          getAnnotation(item.GetLabels(), "pac.weave.works/id"),
+		Name:        getAnnotation(annotations, "policy_name"),
+		PolicyId:    getAnnotation(annotations, "policy_id"),
+		ClusterId:   getAnnotation(annotations, "cluster_id"),
+		Category:    getAnnotation(annotations, "category"),
+		Severity:    getAnnotation(annotations, "severity"),
+		CreatedAt:   item.GetCreationTimestamp().Format(time.RFC3339),
+		Message:     item.Message,
+		Entity:      item.InvolvedObject.Name,
+		Namespace:   item.InvolvedObject.Namespace,
+		ClusterName: clusterName,
+	}
+	if extraDetails {
+		policyValidation.Description = getAnnotation(annotations, "description")
+		policyValidation.HowToSolve = getAnnotation(annotations, "how_to_solve")
+		policyValidation.ViolatingEntity = getAnnotation(annotations, "entity_manifest")
+		err := json.Unmarshal([]byte(getAnnotation(annotations, "occurrences")), &policyValidation.Occurrences)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get occurrences from event: %w", err)
+		}
+		paramsRaw := getAnnotation(annotations, "parameters")
+		if paramsRaw != "" {
+			parameter, err := getPolicyValidationParam([]byte(paramsRaw))
+			if err != nil {
+				return nil, err
+			}
+			policyValidation.Parameters = parameter
+		}
+	}
+	return policyValidation, nil
+}
+
+func getAnnotation(annotations map[string]string, key string) string {
+	value, ok := annotations[key]
+	if !ok {
+		return ""
+	}
+	return value
+}
+
+func getPolicyValidationParam(raw []byte) ([]*pb.PolicyValidationParam, error) {
+	var paramsArr []map[string]interface{}
+	err := json.Unmarshal(raw, &paramsArr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal policy validation parameter, error: %v", err)
+	}
+
+	var parameters []*pb.PolicyValidationParam
+	for i := range paramsArr {
+		param := pb.PolicyValidationParam{
+			Name:     paramsArr[i]["name"].(string),
+			Type:     paramsArr[i]["type"].(string),
+			Required: paramsArr[i]["required"].(bool),
+		}
+
+		if val, ok := paramsArr[i]["config_ref"]; ok {
+			param.ConfigRef = val.(string)
+		}
+
+		val, err := getParamValue(paramsArr[i]["value"])
+		if err != nil {
+			return nil, err
+		}
+		param.Value = val
+		parameters = append(parameters, &param)
+	}
+	return parameters, nil
+}
+
+func getParamValue(in interface{}) (*any.Any, error) {
```

Comment:
I am not very fond of using `in` as an argument name, I would prefer something more descriptive. `in` gives me the feeling like it will be used in a channel or queue.